draft-poidt-teas-actn-poi-assurance.md

```diff
@@ -1,61 +1,195 @@
+They perform single-domain local path computation, when requested by the MDSC. They also perform optical tunnel setup, when requested by the MDSC.
+
+No GMPLS-UNI interaction between IP and Optical equipment is considered.
+This is also the assumption followed in this document: the MDSC performs the function of multi-layer/multi-domain path computation
+through the same mechanisms described in (I-D.ietf-teas-actn-poi-applicability) even if that is applied to the maintenance and protection cases described in the next sections.
+
+{: #yang}
+
+# YANG Data Models for the MPIs
 
+TODO YANG Data Models
+
+Initial set of YANG models that are potentially in the scope of this analysis:
+- ietf-alarms defined in {{!RFC8632}}
+- ietf-performance-monitoring defined in {{!I-D.yu-performance-monitoring-yang}}
+
+{: #optical-network}
+
+# Optical Network Failure and Performance Degradation
+
+{: #optical-fault}
+
+## Fault Detection
+
+TODO Describe fault detection performed by the O-PNC and how this information is reported to the MDSC: see for example the failure scenario in https://github.com/italobusi/draft-poidt-teas-actn-poi-assurance/files/10885907/2023.03.draft-poidt-teas-poi-assurance.pptx (slide 3)
+
+{: #optical-degradation}
+
+## Performance Monitoring
+
+TODO Describe performance monitoring and performance degradation detection performed by the O-PNC and how this information is reported to the MDSC: see for example the degradation scenario in https://github.com/italobusi/draft-poidt-teas-actn-poi-assurance/files/10885907/2023.03.draft-poidt-teas-poi-assurance.pptx (slide 7)
+
+{: #optical-protection}
+
+## Protection Switching
+
+TODO Describe how the MDSC coordinates the protection switching mechanisms at the IP layer (e.g., FRR) and at optical layer, including the reversion when the failure is repaired: see for example the protection switching scenario in https://github.com/italobusi/draft-poidt-teas-actn-poi-assurance/files/10885907/2023.03.draft-poidt-teas-poi-assurance.pptx (slide 3)
+
+{: #optical-maintenance}
+
+## Maintenance
+
+TODO Describe how the MDSC initiates protection switching at the IP layer (e.g., FRR) and at optical layer at the beginning of a maintenance window, including the reversion after the maintenance operations are completed: see for example the maintenance scenario in https://github.com/italobusi/draft-poidt-teas-actn-poi-assurance/files/10885907/2023.03.draft-poidt-teas-poi-assurance.pptx (slide 4)
+
+{: #edge}
+
+# Failures between the IP and Optical edges
+
+{: #edge-fault}
+
+## Fault Detection
+
+TODO Describe the mechanisms to detect when the failure occurs on a router port or on the router node connected with the optical domain: see for example the fault scenarios in https://github.com/italobusi/draft-poidt-teas-actn-poi-assurance/files/10885907/2023.03.draft-poidt-teas-poi-assurance.pptx (slide 5 and slide 6)
```

Comment:
here I think for fault/event detection in optical and IP/MPLS domain we need to have a wider scope. O/P-PNC start to implement alarm filtering capabilities joint with Root Cause Analysis and there the use of RFC8632 for exposing only the results of the RCA at P-PNC and O-PNC North Bound interfaces is not clear. In fact RFC8632 may not be the right data models for that purpose and this should be investigated. There is no real value to flood the MDSC with all Optical and IP/MPLS domains' alarms through the use of RFC8632